tests/test_linear_incident_uptake_model.py

```diff
@@ -156,11 +157,146 @@ def test_project_sequentially():
     y = np.reshape(np.array([0, 1]), (2, 1))
     model.fit(x, y)
 
-    output = iup.LinearIncidentUptakeModel.project_sequentially(
+    output = iup.models.LinearIncidentUptakeModel.project_sequentially(
         elapsed, start, standards, model
     )
 
     assert all(
         np.round(np.delete(output, 0), decimals=10)
         == np.round(np.array(elapsed), decimals=10)
     )
+
+
+def test_trim_outlier_intervals_handles_two_rows(frame):
```

Comment:
I gently disagree with the placement of tests for `trim_outlier_intervals`, `augment_implicit_columns`, and `date_to_whatever` in the "test_linear_incident_uptake_model.py" file. I would keep them in "test_uptake_data.py" because they are operations on data that can be performed regardless of which model, if any, is being built. I understand that the linear incident uptake model is the only model that requires these operations _right now_, but what happens if several types of models require these operations as pre-requisites in the future? In that case, it would seem their tests are misplaced.

Reply:
We agree that the structure of the code should match the structure of the tests: if some method is a method of `Data`, it should be tested in or near the place that other `Data` methods are being tested.

I think we're disagreeing about where to put pre-processing functionality that is currently not shared between models but which could become shared. I lean the other way from you because (1) I put a premium the cleanliness of the top-level `__init__.py` namespace and (2) in my experience it's usually more costly to pre-optimize (e.g., by trying to build generic functionalities for input data manipulation) before seeing what the range of model types is.

Reply:
Yep, that's the conflict. Perhaps I'll come around to your way of thinking, so I'm happy to keep it as is.